### PR TITLE
PCT-1480 - version 2 and 1 of protocol

### PR DESCRIPTION
**Pull request by mpaolino**
This PR contains a refactor in cloud-protocol separating what is our previous version of the protocol, aka v1, and the new version, aka v2. Version 2 only includes the types that are needed for percona-agent refactor, more will be added once other parts of the system are refactored.

### CODE REVIEW

File: proto/v2/instance.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	Subsystem
+	ParentUUID string
+	UUID       string
+	Name       string
+	DSN        string
+	Created    time.Time
+	Deleted    time.Time
+	Properties map[string]string
+	Subsystems []Instance
+}
```
Comment (daniel-nichter) [inline proto/v2/instance.go]: Let's group "high cohesive" structs in the same file. Right now, I have to `vi instance.go subsystem.go system_tree_sync.go` to get a complete picture.

File: proto/v2/tool.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package proto
+
+type ToolInstance struct {
```
Comment (daniel-nichter) [inline proto/v2/tool.go]: What's this for?

File: proto/v2/instance.go (hunk 1 above)
Comment (mpaolino) [inline proto/v2/instance.go]: Sounds good

Comment (mpaolino) [inline proto/v2/instance.go]: this was fixed

**Comment by arvenil** [comment]
:+1:

**Comment by daniel-nichter** [comment]
:+1:

**Pull request merged**